articles/advanced-r-concepts/index.md

```diff
@@ -0,0 +1,333 @@
+
+Let's consider the following example:
+
+```r
+x <- 0:6
+class(x)
+typecast_to_numeric <- as.numeric(x)
+class(typecast_to_numeric).0
+typecast_to_logical <- as.logical(x)
+class(typecast_to_logical)
+typecast_to_char <- as.character(x)
+class(typecast_to_char)
+```
+
+The given code outputs the following:
+
+```txt
+> class(x)
+[1] "integer"
+> typecast_to_numeric <- as.numeric(x)
+> class(typecast_to_numeric)
+[1] "numeric"
+> typecast_to_logical <- as.logical(x)
+> class(typecast_to_logical)
+[1] "logical"
+> typecast_to_char <- as.character(x)
+> class(typecast_to_char)
+[1] "character"
+```
+
+Observe the change in the data types. An application of typecasting makes sense on user-defined data-types. Consider the following applications:
+
+1. Consider a dataframe with a column containing boolean values. If one needs to convert these into integers to train a machine learning model, then typecasting is used. Typecasting converts the boolean into integer data.
+
+2. Supposing one is given a dataframe with one column containing time in seconds. To convert each column of the given dataframe to [time in days format](https://www.rdocumentation.org/packages/PivotalR/versions/0.1.18.3.1/topics/Type%20Cast%20functions), explicit typecasting is used.
+
+For more information on explicit typecasting, refer to this [link](https://study.com/academy/lesson/data-type-conversion-in-r-programming-purpose-functions.html).
+
+### Reading and Writing Data
+
+R provides many functions for reading and writing data. There are mainly three data formats used in R. They are `.txt,` `.csv,` and `.RData.` The functions are defined for each of the file-types and are given as follows:
+
+1. read.table: `read.table("test.txt")`
+2. read.csv: `read.csv("test.csv")`
+3. load: `load("file_name.RData")`
+
+For writing into files, R provides the following functions:
+
+1. write.table: `write.table(data, file='file_name.txt')`
+2. write.csv: `write.csv(data, file='file_name.csv')`
+3. save: `save(x,y,"file_name.Rdata")`
+
+Whether reading from or writing to files, ensure that the file directory is correctly mentioned. Relative paths are used extensively in file handling. Consider going through this [answer](https://stackoverflow.com/questions/36834767/how-to-use-rstudio-relative-paths) on StackOverflow for a glance on relative paths.
+
+### Accessing inbuilt Datasets
+At this point, you are in a position to use R effectively. While learning various machine learning algorithms, the best place to start is the in-built datasets. These are small in size and are cleaned thoroughly.
+
+These datasets were the benchmark datasets around a decade ago. Today, budding data scientists use these as tools to learn the various machine learning algorithms. A few of the datasets that R provides are as follows:
```

Comment:
Explain what this means. How do data scientists use or import these datasets? What is the minimum the reader needs to know?